iroh-net/src/magicsock/endpoint.rs

```diff
@@ -1112,6 +1193,15 @@ impl PeerMap {
         self.by_ip_port.insert(*ipp, id);
     }
 
+    /// Saves the known peer info to the given path, returning the number of peers persisted.
+    pub(super) fn save_to_file(&self, path: &Path) -> anyhow::Result<usize> {
+        let known_peers = self.known_peers();
+        let count = known_peers.len();
+        let serialized = postcard::to_stdvec(&known_peers)?;
+        std::fs::write(path, serialized)?;
```

Comment:
should this be async?

Reply:
Also, we shouldn't directly write to the path, but rather to a `tmp` file (fine in the same location) and rename when we have successfully written, to avoid having errors during writing leaving corrupted files behind

Reply:
didn't get to do this but will look into it